src/main/java/com/jfrog/ide/idea/ui/webview/WebviewObjectConverter.java

```diff
@@ -206,4 +205,24 @@ private static Reference[] convertReferences(List<String> xrayReferences) {
             return new Reference(parts[1].substring(0, parts[1].length() - 1), parts[0].substring(1));
         }).toArray(Reference[]::new);
     }
+
+    private static String getPackageTypeName(String componentId) {
+        String GENERIC_PKG_TYPE = "Generic";
+        String[] compIdParts = componentId.split("://");
```

Comment:
You can use `StringUtils.substringBefore(componentId, "://");` to get the component prefix.

Reply:
But I won't know if the "://" separator exists. There are cases when it doesn't, and Xray refers to the project as Generic.

Reply:
In this case the `substringBefore` will return an empty string and you can return "Generic"